### DISCUSSION TITLE
[QUESTION]Why is the operator of computation slower when computation overlaps with communication

### DISCUSSION THREAD

**Discussion by yu-depend** [NVIDIA/Megatron-LM#1021] [Q&A]
Why is the operator of computation slower when computation overlaps with communication

**Comment by elliottnv** [comment]
This question is too general. Can you provide more details?

**Comment by yu-depend** [comment]
When `-- overlap-grad-Reduce `is used, the compute kernel with reduce_scatter overlap (the duration of GEMM kernel with red rectangle in the figure is longer than that of kernel without overlap). I assume that the NCCL kernel seized the SM resource?Please explain it.
![megatron-lm-issue](https://github.com/user-attachments/assets/2d8e1eb4-198d-4b6b-8746-65ab352d5109)

**Comment by deepakn94** [comment]
Right, there is some resource contention. This is expected.

**Reply by yu-depend** [reply]
Thank you for your reply.
